src/components/BasicTableModal/index.jsx

```diff
@@ -91,24 +99,38 @@ useEffect(() => {
 
   const handleSaveClick = async (rowId) => {
     try {
-      const editedRow = editedRows.find((row) => row.id === rowId);
+      const editedRowIndex = editedRows.findIndex((row) => row.id === rowId);
+
+          if (editedRowIndex !== -1) {
+            const editedRow = { ...editedRows[editedRowIndex] };
+            const { productName, quantity, expiration, batchNumber, actualQuantity } = editedRow;
+
+            const quantityDiff = parseInt(actualQuantity, 10) - parseInt(quantity, 10)  ;
+
 
-      if (editedRow) {
-        const { productName, quantity, actualQuantity, expiration, batchNumber } = editedRow;
+            editedRow.quantity = quantityDiff.toString();
 
-        const response = await saveEditedQuantity(
-          productName,
-          quantity,
-          actualQuantity,
-          expiration,
-          batchNumber,
-          stockRoom.results[0]?.uuid
-        );
 
-        if (response && response.ok) {
-          setSaveClicked(true);
+            const updatedRows = [...editedRows];
+            updatedRows[editedRowIndex] = editedRow;
+
+            const response = await saveEditedQuantity(
```

Comment:
Should we have a check for quantityDiff not equal to zero before calling the api?